tests/common.py

```diff
@@ -143,6 +143,23 @@ def reset(self, seed=None, options=None):
         return self.observation_space.sample(), {}
 
 
+class DummyTextEnv(gym.Env):
+    def __init__(self):
+        self.action_space = spaces.Text(max_length=10, min_length=2, charset="01")
+
+        self.observation_space = spaces.Text(max_length=20)
```

Comment:
You should probably set `Text(max_length=20, charset=string.printable)` as that will be a list of all the printable ascii chars (length 100) rather than just the letters a-z and digits which is the default for gymnasium. You could have the charset be all of the utf-8 characters but that would be a million (as all unicode chars can be encoded) so sampling from that will probably take a while and not be ideal.
If you want the printable unicode multilingual characters (65,000) length it would be `utf8_characters = [chr(i) for i in range(0x10000) if unicodedata.category(chr(i)) != "Cc"]`, but that's probably not necessary besides maybe showing people that it can be done.

Reply:
Trying with all unicode it fails with an informative error `UnicodeEncodeError: 'utf-8' codec can't encode character '\udb25' in position 2: surrogates not allowed`

Reply:
We need to somehow create a charset for all legal unicode characters, and decide what our stance is on surrogate characters, this may be nontrivial. Naively, I tried:

````
utf8_charset = "".join([chr(i) for i in range(sys.maxunicode)])
````
and I got a similar error
````
E   UnicodeEncodeError: 'utf-8' codec can't encode character '\ud937' in position 3: surrogates not allowed
````

Here is a  h5py doc page talking about surrogate characters:
https://docs.h5py.org/en/3.2.1/strings.html

Reply:
I don't understand what you mean with "decide what our stance is on surrogate characters"